src/libraries/Common/src/System/Security/Cryptography/MLDsaCng.Windows.cs

```diff
@@ -35,17 +37,33 @@ internal MLDsaCng(CngKey key, bool transferOwnership)
             _key = key;
         }
 
+        [SupportedOSPlatform("windows")]
         private static partial MLDsaAlgorithm AlgorithmFromHandle(CngKey key, out CngKey duplicateKey)
         {
+#if !NETFRAMEWORK
+            // Non-Windows call in Microsoft.Bcl.Cryptography can get here if MLDsaCng was type-forwarded
```

Comment:
I'm not sure what type-forwards have to do with it.

If this code is running inside Microsoft.Bcl.Cryptography, it wasn't type-forwarded.  It forwards from M.B.C to System.Security.Cryptography for .NET 10+.

Assuming that we have this class in Microsoft.Bcl.Cryptography for .NET 8 and .NET 9 on Windows access, then type forwards aren't involved when it's getting called.  Then it's down to "how'd you get here?".  Assuming the type isn't public (since we previously said it shouldn't be) it means that we called cert.GetMLDsaPrivateKey (since PublicKey would use the BCrypt version)... which really means this check belongs out there.  But, either way, type forwards aren't involved...

Reply:
Moved check out to common class.